amida-tech/recruitment-registry
# Feature/rr 913 role based consent

If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### What does this PR do? This implements role based consents

#### Related JIRA tickets: RR-913

#### How should this be manually tested?

1) POST /consent-types with a new field 'role'. It can be either 'participant' or 'clinician'.
2) [GET] /user-consent-documents. Check that either role does not get other's consent document.
3) Use boolean query parameter 'role-only' to show only role based consent documents with [GET] /user-consent-documents. This can be used after login to show unsigned role based consent.
4) You should not able to [POST] /survey-consents with a role consent type.
5) SAGE endpoints (/consents) are also updated similarly in case you want to test. I do not believe these are currently being used. 

#### Background/Context NA

#### Screenshots (if appropriate): NA
